src/sys/unix.rs

```diff
@@ -2154,6 +2157,56 @@ impl crate::Socket {
             )
         }
     }
+
+    /// Get the value of the `TCP_CONGESTION` option for this socket.
+    ///
+    /// For more information about this option, see [`set_tcp_congestion`].
+    ///
+    /// [`set_tcp_congestion`]: Socket::set_tcp_congestion
+    #[cfg(all(feature = "all", any(target_os = "freebsd", target_os = "linux")))]
+    #[cfg_attr(
+        docsrs,
+        doc(cfg(all(feature = "all", any(target_os = "freebsd", target_os = "linux"))))
+    )]
+    pub fn tcp_congestion(&self) -> io::Result<Vec<u8>> {
+        let mut payload: MaybeUninit<[u8; TCP_CA_NAME_MAX]> = MaybeUninit::uninit();
+        let mut len = size_of::<[u8; TCP_CA_NAME_MAX]>() as libc::socklen_t;
```

Comment:
This should really use the length of the payload. Although that's the same.
```suggestion
        let mut len = payload.len();
```